mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ServerState.scala

```diff
@@ -953,6 +924,7 @@ import scala.util.{Failure, Success}
   // State event handling
 
   def prepareServerUnpublisher(data: Start): Behavior[Event] = Behaviors.setup { context =>
+    // FIXME using Promise for replies seems strange. I would prefer ActorRef and messages.
     val reply = Promise[RemotePacketRouter.Registered.type]
     data.packetRouter ! RemotePacketRouter.Register(context.self.unsafeUpcast, data.clientId, data.packetId, reply)
```

Comment:
@patriknw I’m having a further think on the use of promises for ask patterns within types actors. The convention is to wrap actor reply messages, but is this really better than passing in a promise for a reply?

The only downside to the promise that I can see is that one has to attempt to fail it after a timeout given no success (so that resources are cleared up).

The upside is the avoidance of having to create a wrapped message type.